### ISSUE TITLE
Socket for clamd not found yet

### ISSUE THREAD

**Issue by EmadBinAbid** [Cisco-Talos/clamav#760]
The two previously reported issues are https://github.com/Cisco-Talos/clamav/issues/754 and https://github.com/Cisco-Talos/clamav/issues/757, I tried following the same config, but I'm still not able to bypass this error `Socket for clamd not found yet, retrying`.

I am mounting the following config.

**Freshclam Config**
```
# Use freshclamConfig to define the content of /etc/clamav/freshclam.conf. Example:
freshclamConfig: |
  ###############
  # General
  ###############

  DatabaseDirectory /data
  UpdateLogFile /dev/stdout
  LogTime yes
  PidFile /var/run/clamav/freshclam.pid
  DatabaseOwner root

  ###############
  # Updates
  ###############

  DatabaseMirror database.clamav.net
  ScriptedUpdates yes
  NotifyClamd /etc/clamav/clamd.conf
  Bytecode yes
```

**Clamd Config**
```
clamdConfig: |
  DatabaseDirectory /data
  TemporaryDirectory /tmp
  LogTime yes
  PidFile /var/run/clamav/clamd.pid

  # The Freshclam process in clamav/clamav looks for the Clamd socket file at path /run/clamav/clamd.sock
  LocalSocket /run/clamav/clamd.sock
  TCPSocket 3310
  Foreground yes

  ###############
  # Results
  ###############

  DetectPUA yes
  ExcludePUA NetTool
  ExcludePUA PWTool
  AlgorithmicDetection yes
  Bytecode yes

  ###############
  # Scan
  ###############

  ScanPE yes
  DisableCertCheck yes
  ScanELF yes
  ScanOLE2 yes
  ScanPDF yes
  ScanSWF yes
  ScanMail yes
  PhishingSignatures yes
  PhishingScanURLs yes
  ScanHTML yes
  ScanArchive yes

  ###############
  # Scan
  ###############

  MaxScanSize 150M
  MaxFileSize 30M
  MaxRecursion 10
  MaxFiles 15000
  MaxEmbeddedPE 10M
  MaxHTMLNormalize 10M
  MaxHTMLNoTags 2M
  MaxScriptNormalize 5M
  MaxZipTypeRcg 1M
  MaxPartitions 128
  MaxIconsPE 200
  PCREMatchLimit 10000
  PCRERecMatchLimit 10000

  # Our attachment limits are 100MB
  StreamMaxLength 100M
```

**Container Logs**
```
Starting Freshclamd
Starting ClamAV
Socket for clamd not found yet, retrying (0/1800) ...Thu Nov 17 10:51:58 2022 -> --------------------------------------
Thu Nov 17 10:51:58 2022 -> freshclam daemon 0.105.1 (OS: Linux, ARCH: x86_64, CPU: x86_64)
Thu Nov 17 10:51:58 2022 -> ClamAV update process started at Thu Nov 17 10:51:58 2022
Thu Nov 17 10:51:58 2022 -> ClamAV update process started at Thu Nov 17 10:51:58 2022
Thu Nov 17 10:51:58 2022 -> daily.cvd database is up-to-date (version: 26723, sigs: 2011219, f-level: 90, builder: raynman)
Thu Nov 17 10:51:58 2022 -> daily.cvd database is up-to-date (version: 26723, sigs: 2011219, f-level: 90, builder: raynman)
Thu Nov 17 10:51:58 2022 -> main.cvd database is up-to-date (version: 62, sigs: 6647427, f-level: 90, builder: sigmgr)
Thu Nov 17 10:51:58 2022 -> main.cvd database is up-to-date (version: 62, sigs: 6647427, f-level: 90, builder: sigmgr)
Thu Nov 17 10:51:58 2022 -> bytecode.cvd database is up-to-date (version: 333, sigs: 92, f-level: 63, builder: awillia2)
Thu Nov 17 10:51:58 2022 -> bytecode.cvd database is up-to-date (version: 333, sigs: 92, f-level: 63, builder: awillia2)
Thu Nov 17 10:51:58 2022 -> --------------------------------------
Socket for clamd not found yet, retrying (54/1800) ...%  
```

After that, the pod gets restarted and eventually rests into `CrashLoopBackOff` state.

Can anyone clarify? 

@micahsnyder

**Comment by micahsnyder** [comment]
@EmadBinAbid I trust it is working for you now with the latest image?

**Comment by EmadBinAbid** [comment]
@micahsnyder, yeah, it seems to be working fine now - will let you know if I face any further issues.

**Comment by dangeruss23** [comment]
I was receiving the same error message, but it was not resolved when I upgraded to the latest image.  For me, the fix was in the LocalSocket setting in clamd.conf.

I had initially set it to this:

LocalSocket /run/clamav/clamd.socket

I was able to scan files successfully, but errors were being logged that the "Socket for clamd not found yet, retrying" and the clamav container would restart every 30 minutes.  The issue was resolved by changing it to this:

LocalSocket /run/clamav/clamd.sock

**Comment by kbdavis07** [comment]
"I was able to scan files successfully, but errors were being logged that the "Socket for clamd not found yet, retrying" and the clamav container would restart every 30 minutes. The issue was resolved by changing it to this:

LocalSocket /run/clamav/clamd.sock"

For me on Azure AKS, changing it to:

LocalSocket /run/clamav/clamd.sock

Did not work, what other issue could this be in an AKS env?

Locally no issues just when deploy to AKS have issue.
